R/valley.R

```diff
@@ -0,0 +1,330 @@
+#' 
+#' @param asset_urls a list of STAC records to be retrieved
+#' @param bb A bounding box (compliant with CRiSp, i.e. as a matrix with 4 elements: xmin, ymin, xmax, ymax)
+#' 
+#' @return A a merged dem fromm retrieved assets cropped to the bounding box
+#' @export
+get_stac_dem <- function(bb, asset_urls){
+    dem <- asset_urls |>
+    lapply(rast) |>
+    lapply(crop, as.vector(t(bb))) |>
+    do.call(merge, args=_)
+}
+
+#' Load local dem from file. Optionally crop to a bounding box.
+#' NOTE: User is responsible for dem and bounding box specifications being compatible
+#' 
+#' @param demfilepath path to dem to be loaded
+#' @param bb A bounding box (compliant with CRiSp, i.e. as a matrix with 4 elements: xmin, ymin, xmax, ymax). Optional
+#' 
+#' @return A a merged dem from cropped to the bounding box
+#' @export
+get_local_dem <- function(bb, demfilepath){
+    dem <- 
+}
+
+#'Load dem either from file or a STAC endpoint
+#' 
+#' @param resource from whcih to source dem. Can be "STAC" or "FILE".
+#' if "STAC" the parameters the following parameters must/can be supplied as named parameters
+#' @param endpoint STAC endpoint to use (required)
+#' @param collection STAC collection to use (required)
+#' @param limit limit of records to retrieve (optional)
+#' if resource is "FILE" the filepath to the dem must be specified using the named parameter
+#' @param demfilepath
+#' 
+#' @return dem 
+#' @export
+get_dem <- function(bb, resource="STAC",...){
+    kwargs <- list(...)
+    if (resource == "STAC"){
+        if(length(kwargs) && !is.null(...)){
+            endpoint = kwargs$endpoint
+            collection = kwargs$collection
+            if("limit" in names(kwargs)){
+                limit=kwargs$limit
+                asset_urls <- get_stac_asset_urls(bb,endpoint=endpoint,collection=collection,limit=limit)
+            } else {
+                asset_urls <- get_stac_asset_urls(bb,endpoint=endpoint,collection=collection)
+            }
+        } else {
+            asset_urls <- get_stac_asset_urls(bb)
+        }
+        dem <- get_stac_dem(bb, asset_urls)
+    } else if (resource == "FILE") {
+        demfilepath = kwargs$demfilepath
+        dem <- get_local_dem(bb, demfilepath)
+    } else {
+        #add error statement
+    }
+}
+
+
+#'Write dem to cloud optimized GeoTiff file as specified location
+#' 
+#' @param dem to write to file
+#' @param fpath filepath for output. If no output directory is specified (see below) fpath is parsed to determine
+#' the output directory
+#' @param output_directory where file should be written. If specified fpath is treated as filename only.
+#' 
+#' @export
+dem_to_COG <- function(dem,fpath,output_directory=None,){
+    if (output_directory == None){
+        file_name = basename(fpath)
+        directory_name = dirname(file)
+    } else {
+        file_name = fpath
+        directory_name = output_directory
+    }
+    data_dir  <- here(directory_name)
```

Comment:
I agree. `fpath` should be enough as a single argument for path information. `here::here()` is indeed used to handle different situations and for reproducibility, often because Windows and macOS/Linux construct paths in different ways.